Common/AR/source/Android/ArCoreWrapper.cpp

```diff
@@ -0,0 +1,933 @@
+
+  int32_t xImageSize = 0;
+  int32_t yImageSize = 0;
+  ArCameraIntrinsics_getImageDimensions(m_arSession, cameraIntrinsics, &xImageSize, &yImageSize);
+
+  ArCameraIntrinsics_destroy(cameraIntrinsics);
+
+  return
+  {
+    static_cast<double>(xFocalLength), static_cast<double>(yFocalLength),
+    static_cast<double>(xPrincipal), static_cast<double>(yPrincipal),
+    static_cast<double>(xImageSize), static_cast<double>(yImageSize)
+  };
+}
+
+/*!
+  \internal
+  Implementation based on the documentation: https://developers.google.com/ar/reference/c/group/frame#arframe_hittest
+ */
+std::array<double, 7> ArCoreWrapper::hitTest(int x, int y) const
+{
+  if (!m_arSession || !m_arCamera)
+    return {};
+
+  // create hit result list
+  ArHitResultList* hitResults = nullptr;
+  ArHitResultList_create(m_arSession, &hitResults);
+  if (!hitResults)
+    return {};
+
+  // try to find the location point
+  ArFrame_hitTest(m_arSession, m_arFrame, x, y, hitResults);
+
+  int32_t size = 0;
+  ArHitResultList_getSize(m_arSession, hitResults, &size);
+  if (size <= 0)
+  {
+    ArHitResultList_destroy(hitResults);
+    return {};
+  }
+
+  ArHitResult* hitResult = nullptr;
+  ArHitResult_create(m_arSession, &hitResult);
+  if (!hitResult)
+  {
+    ArHitResultList_destroy(hitResults);
+    return {};
+  }
+
+  ArHitResultList_getItem(m_arSession, hitResults, 0, hitResult);
+
+  ArPose* pose = nullptr;
+  ArPose_create(m_arSession, nullptr, &pose);
+  if (!hitResult)
+  {
+    ArHitResult_destroy(hitResult);
+    ArHitResultList_destroy(hitResults);
+    return {};
+  }
+
+  ArHitResult_getHitPose(m_arSession, hitResult, pose);
+
+  float poseRaw[7] = {};
+  ArPose_getPoseRaw(m_arSession, pose, poseRaw);
+
+  // destroys objects
+  ArPose_destroy(pose);
+  ArHitResult_destroy(hitResult);
+  ArHitResultList_destroy(hitResults);
+
+  return { poseRaw[0], poseRaw[1], poseRaw[2], poseRaw[3], poseRaw[4], poseRaw[5], poseRaw[6] };
+}
+
+/*!
+  \internal
+ */
+std::array<float, 8> ArCoreWrapper::transformedUvs() const
+{
+  return m_transformedUvs;
+}
+
+/*!
+  \internal
+  properties for debug mode
+ */
+QColor ArCoreWrapper::pointCloudColor() const
+{
+  if (m_arCorePointCloudRenderer)
+    return m_arCorePointCloudRenderer->pointCloudColor();
+  return QColor();
+}
+
+/*!
+  \internal
+ */
+void ArCoreWrapper::setPointCloudColor(const QColor& pointCloudColor)
+{
+  if (pointCloudColor.isValid())
+  {
+    if (!m_arCorePointCloudRenderer)
+    {
+      m_arCorePointCloudRenderer.reset(new ArCorePointCloudRenderer(this));
+      m_arCorePointCloudRenderer->initGL();
+    }
+
+    m_arCorePointCloudRenderer->setPointCloudColor(pointCloudColor);
+  }
+  else
+  {
+    m_arCorePointCloudRenderer.reset();
+  }
+}
+
+/*!
+  \internal
+ */
+int ArCoreWrapper::pointCloudSize() const
+{
+  if (m_arCorePointCloudRenderer)
+    return m_arCorePointCloudRenderer->pointCloudSize();
+  return -1;
```

Comment:
Add a line after the if-statement return. If there are more of these, I won't comment on them all; please fix them.

Edit:
>If there are more of these,

There are many more...